### DISCUSSION TITLE
🎉 Tech preview of Blynk.Edgent for MicroPython

### DISCUSSION THREAD

**Discussion by vshymanskyy** [micropython/micropython#15864] [General]
🚀  The tech preview of Blynk.Edgent for MicroPython has been released 🚀
https://github.com/Blynk-Technologies/Blynk-MicroPython-Edgent

**This release includes:**
- Smartphone Apps
- Web Dashboards
- Network management for WiFi, Ethernet and Cellular
- Secure MQTT connection to the cloud
- OTA updates using Blynk.Console
- Time Zone handling (including DST transitions), Sunrise/Sunset calculation
- Configuration storage and management
- and many [other features](https://github.com/Blynk-Technologies/Blynk-MicroPython-Edgent?tab=readme-ov-file#features) on top of (almost) vanilla MicroPython

This is available for Generic boards like **ESP32 and RPi Pico W**.
With the next releases, we'll introduce pre-configured builds for specific dev boards.

Feedback is welcome!

**Comment by bidrohini** [comment]
Does it support all the versions of the ESP32?

**Reply by vshymanskyy** [reply]
Yes, it does support: [ESP32, ESP32-S3, ESP32-C3](https://github.com/Blynk-Technologies/Blynk-MicroPython-Edgent/releases/latest).
ESP32-C6 support will be added as it gets merged into MicroPython

**Comment by vshymanskyy** [comment]
Ultimately, I'd like to ship it in a form of MicroPython library package.
It will require upstreaming our changes to MicroPython, which will probably take some time.
So for now we're providing ready-to-use builds.
